pkg/routeagent/controllers/route/route.go

```diff
@@ -541,20 +548,31 @@ func (r *Controller) handleRemovedEndpoint(obj interface{}) {
 		r.gwVxLanMutex.Lock()
 		defer r.gwVxLanMutex.Unlock()
 
-		klog.V(6).Infof("Endpoint matches the cluster ID of this cluster")
-		err := r.vxlanDevice.deleteVxLanIface()
-		if err != nil {
-			klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %v", err)
-			return
+		if object.Spec.Hostname == hostname {
+			klog.Infof("Local Gateway Endpoint (IP %s) removed: deleting vxlan interface",
+				object.Spec.PrivateIP)
+		} else {
+			klog.Infof("Gateway Endpoint (IP %s) for our cluster removed: deleting vxlan interface",
+				object.Spec.PrivateIP)
+		}
+
+		if r.vxlanDevice == nil {
+			klog.Errorf("r.vxlanDevice == nil , this should never happen, endpoint creation event missed?.")
+		} else {
+			err := r.vxlanDevice.deleteVxLanIface()
+			if err != nil {
+				klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %s", err.Error())
+				return
+			}
+			r.vxlanDevice = nil
 		}
-		r.vxlanDevice = nil
```

Comment:
I suggest this structure:

```
if object.Spec.ClusterID != r.clusterID {
    return
}
	
r.gwVxLanMutex.Lock()
defer r.gwVxLanMutex.Unlock()
	
if r.vxlanDevice == nil {
    klog.Errorf("vxlanDevice is not set - ignoring removed endpoint %q", object.Name)
    return
}

if r.isGatewayNode {
    klog.Infof("Gateway Endpoint %q with IP %q was removed: deleting vxlan interface", object.Name, object.Spec.PrivateIP)
    r.cleanRoutes()
} else {
    klog.Infof("Non-gateway Endpoint %q with IP %q was removed: deleting vxlan interface", object.Name, object.Spec.PrivateIP)
}

err = r.vxlanDevice.deleteVxLanIface()
r.vxlanDevice = nil
if err != nil {
    klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %v", err)
} 
```

Reply:
I don't feel comfortable making "structural" changes before a release, I've been bitten by unexpected things in the past. I just want to cleanup the logging as much as I can. Do you think it's fine if we do this after release?

Reply:
Yeah, I think it is... I'm just pre-release-scared ;)

But a separate PR may be better, as this is more about the logs, I guess.